src/k8s-engine/wsl.ts

```diff
@@ -390,15 +390,33 @@ export default class WSLBackend extends events.EventEmitter implements K8s.Kuber
     return this.#desiredPort;
   }
 
+  /**
+   * List the registered WSL2 distributions.
+   */
   protected async registeredDistros({ runningOnly = false } = {}): Promise<string[]> {
-    const args = ['--list', '--quiet'];
+    const stdout = await this.execWSL({ capture: true }, '--list', '--verbose');
+    // As wsl.exe may be localized, don't check state here.
+    const parser = /^[\s\*]+(?<name>.*?)\s+\w+\s+(?<version>\d+)\s*$/;
+
+    let result = stdout.trim()
+      .split(/[\r\n]+/)
+      .slice(1) // drop the title row
+      .map(line => line.match(parser))
+      .filter(defined)
+      .filter(result => result.groups?.version === '2')
+      .map(result => result.groups?.name)
+      .filter(defined);
 
     if (runningOnly) {
-      args.push('--running');
+      const runningOut = await this.execWSL({ capture: true }, '--list', '--quiet', '--running');
```

Comment:
I assume this is more a locale-proof way of getting the running WSLs. Otherwise I think you could do something like this:
```
const parser = /^[\s*]+(?<name>.*+)\s+(?<state>\w+)\s+(?<version>\d+)\s*$/;
…
      .filter(result => result.groups?.version === '2')
      .filter(result => !runningOnly || result.groups?.state === 'Running')
      .map(result => result.groups?.name); // Assuming the change in the regex
```
and then no need to run the second `wsl --list...` process.

Reply:
Yeah, exactly — I have no confidence it parsing the actual text in the listing. Hence the `// As wsl.exe may be localized, don't check state here.` comment.

Reply:
Right, I missed that.